solidity/test/ContractTest.t.sol

```diff
@@ -118,6 +118,21 @@ contract E2EMockContractTest_Set_Mapping_Vars is CommonE2EBase {
   // _contractTest.set_uint256ToAddressToBytes32(1, _owner, bytes32('4'));
   // assertEq(_contractTest.uint256ToAddressToBytes32(1, _owner), bytes32('4'));
 
+  function test_Set_Uint256ToMyStructMappings() public {
+    _contractTest.set_uint256ToMyStruct(1, IContractTest.MyStruct(100, 'hundred'));
+    (uint256 _value, string memory _name) = _contractTest.uint256ToMyStruct(1);
+    assertEq(_value, 100);
+    assertEq(_name, 'hundred');
+  }
+
+  function test_Set_Uint256ToMyNestingStructMappings() public {
+    _contractTest.set_uint256ToMyNestingStruct(1, IContractTest.MyNestingStruct(100, IContractTest.MyNestedStruct(200)));
+    (uint256 _someField, IContractTest.MyNestedStruct memory _myNestedStruct) =
+      _contractTest.uint256ToMyNestingStruct(1);
+    assertEq(_someField, 100);
+    assertEq(_myNestedStruct.nestedField, 200);
```

Comment:
Opposing what [BES-125](https://linear.app/defi-wonderland/issue/BES-125/mappings-with-enumerablesetaddressset-as-value) states, nested structs don't cause compilation errors and even get mocked correctly.

However, the error message displayed in the task is more precise than its assumptions:

> Types containing (nested) mappings can only be parameters or return variables of internal or library functions.

Reply:
Regarding the solution, the current setup retrieves data through the `ast` object, which doesn't contain detailed struct info as the `abi` object does — so mappings being nested by a struct cannot be checked.

This issue may be the reason why [BES-128](https://linear.app/defi-wonderland/issue/BES-128/look-for-alternatives-to-ast-for-parsing-contracts) exists.